packages/wallets/src/components/WalletsCarousel/WalletsCarousel.tsx

```diff
@@ -9,57 +8,54 @@ import './WalletsCarousel.scss';
 
 const WalletsCarousel: React.FC = () => {
     const [isWalletSettled, setIsWalletSettled] = useState(true);
-    const [showWalletsCarouselHeader, setShowWalletsCarouselHeader] = useState(false);
-    const [heightFromTop, setHeightFromTop] = useState(0);
+    const [hideWalletsCarouselHeader, setHideWalletsCarouselHeader] = useState(false);
+    const contentRef = useRef(null);
     const { data: activeWallet, isLoading: isActiveWalletLoading } = useActiveWalletAccount();
 
-    const containerRef = useRef<HTMLDivElement>(null);
-
-    // function to handle scrolling event for hiding/displaying WalletsCarouselHeader
-    // walletsCarouselHeader will be displayed when height from top of screen is more than 100px
-    const handleScroll = useCallback(() => {
-        if (containerRef.current) {
-            const newHeightFromTop = containerRef.current.getBoundingClientRect().top;
-            setHeightFromTop(newHeightFromTop);
-            heightFromTop && setShowWalletsCarouselHeader(heightFromTop < -100);
-        }
-    }, [heightFromTop]);
-
-    //this handle scroll function listens to the scroll as well as touchmove events to handle drag scrolling on mobile
-    useEventListener('touchmove', handleScroll, containerRef);
-    useEventListener('touchend', handleScroll, containerRef);
-    useEventListener('scroll', handleScroll, containerRef);
-    useEventListener('wheel', handleScroll, containerRef);
-
+    // useEffect hook to handle event for hiding/displaying WalletsCarouselHeader
+    // walletsCarouselHeader will be displayed when WalletsCarouselContent is almost out of viewport
     useEffect(() => {
-        let isMounted = true;
+        const observer = new IntersectionObserver(
+            entries => {
+                entries.forEach(entry => {
+                    setHideWalletsCarouselHeader(entry.isIntersecting);
+                });
+            },
+            { threshold: 0.4 } // triggers when 40% of the element is left in view
+        );
+
+        const currentContentRef = contentRef.current;
 
-        if (isMounted) {
-            handleScroll();
+        if (currentContentRef) {
+            observer.observe(currentContentRef);
         }
 
         return () => {
-            isMounted = false;
+            if (currentContentRef) {
+                observer.unobserve(currentContentRef);
+            }
         };
-    }, [handleScroll, heightFromTop]);
+    }, [contentRef]);
```

Comment:
hey @lubega-deriv 
```suggestion
    }, []);
```
believe that there is no sense to put `contentRef` into deps of the useEffect because the reference to `contentRef` never changes between the rernders, so `contentRef` deps here is redundant